### PR TITLE
10 view monthly summary report feature

### PR DESCRIPTION
**Pull request by Zeinab979**

### CODE REVIEW

File: src/test/kotlin/MonthlyReportTest.kt [hunk 1]
```diff
@@ -0,0 +1,141 @@
+import org.example.data.datasource.InMemoryDatasource
+import org.example.data.models.Category
+import org.example.data.models.Report
+import org.example.data.models.Transaction
+import org.example.data.models.TransactionType
+import java.time.LocalDate
+
+fun should_generateMonthlySummary_when_transactionsExist() {
+    val datasource = InMemoryDatasource()
```
Comment (devsadeq) [inline src/test/kotlin/MonthlyReportTest.kt]: We should use repository not data source

File: src/test/kotlin/MonthlyReportTest.kt [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    // Then
+    val expectedReport = Report(
+        totalIncome = 0.0,
+        totalExpense = 0.0,
+        netBalance = 0.0
+    )
+    check(actual = report, expected = expectedReport)
+}
+
+fun should_generateCorrectSummary_whenMultipleTransactionsExist() {
```
Comment (devsadeq) [inline src/test/kotlin/MonthlyReportTest.kt]: We can replace the first test case with this one

File: src/test/kotlin/MonthlyReportTest.kt [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    val expectedReport = Report(
+        totalIncome = 3500.0,
+        totalExpense = 1500.0,
+        netBalance = 2000.0
+    )
+    check(actual = report, expected = expectedReport)
+}
+fun should_generateEmptyMonthlySummary_when_noTransactionsExist(){
+    val datasource = InMemoryDatasource()
+    //Null Given
```
Comment (devsadeq) [inline src/test/kotlin/MonthlyReportTest.kt]: this is not null GIVEN , plus add space at the start of the comment

File: src/test/kotlin/MonthlyReportTest.kt [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    //Then
+    val expectedReport = Report(
+        totalIncome = 3500.0,
+        totalExpense = 1500.0,
+        netBalance = 2000.0
+    )
+    check(actual = report, expected = expectedReport)
+}
+fun should_generateEmptyMonthlySummary_when_noTransactionsExist(){
+    val datasource = InMemoryDatasource()
```
Comment (devsadeq) [inline src/test/kotlin/MonthlyReportTest.kt]: In all test cases we need to use the repository not the data source

File: src/test/kotlin/MonthlyReportTest.kt [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+    //When
+    val report = datasource.getMonthlyReport(LocalDate.of(2025,4,1))
+    //Then
+    val expectedReport = Report(
+        totalExpense = 0.0,
+        totalIncome = 0.0,
+        netBalance = 0.0
+    )
+    check(actual = report, expected = expectedReport)
+}
```
Comment (devsadeq) [inline src/test/kotlin/MonthlyReportTest.kt]: Create one more test case for the negative net balance

File: src/test/kotlin/TestRunner.kt [hunk 6]
```diff
@@ -12,6 +12,14 @@ fun main() {
         shouldAddTransactionWhenDescriptionIsNull()
         shouldNotAddTransactionWhenDateIsInTheFuture()
         //endregion
+
+        //monthly report testcases
```
Comment (devsadeq) [inline src/test/kotlin/TestRunner.kt]: change this comment to region comment

File: src/main/kotlin/repository/TransactionRepositoryImpl.kt [hunk 7]
```diff
@@ -27,7 +27,7 @@ class TransactionRepositoryImpl(private val datasource: LocalDatasource) : Trans
     }
 
     override fun getMonthlyReport(date: LocalDate): Report {
```
Comment (devsadeq) [inline src/main/kotlin/repository/TransactionRepositoryImpl.kt:29]: Why this function is not used?

File: src/main/kotlin/data/datasource/InMemoryDatasource.kt [hunk 8]
```diff
@@ -38,7 +39,23 @@ class InMemoryDatasource : LocalDatasource {
     }
 
     override fun getMonthlyReport(date: LocalDate): Report {
```
Comment (devsadeq) [inline src/main/kotlin/data/datasource/InMemoryDatasource.kt:58]: Better formating for this function

**Review by devsadeq — CHANGES_REQUESTED** [review]
